PhotoPlusBackend/photoplus/src/main/java/pl/polsl/photoplus/security/custom/CustomBasicAuthenticationFilter.java

```diff
@@ -59,21 +62,26 @@ private Authentication getLoginPasswordAuthentication(final HttpServletRequest r
         if (token == null) {
             return null;
         }
-        final DecodedJWT decodedJwt = JWT.require(Algorithm.HMAC256(modelPropertiesService.securitySecret.getBytes()))
-                .build()
-                .verify(tokenWithoutPrefix);
-        final String login = decodedJwt.getSubject();
-        if (login != null && isNotLoggedOut(tokenWithoutPrefix)) {
-            final Optional<User> foundUser = userRepository.findUserByLogin(login);
-            if (foundUser.isPresent()) {
-                final UsernamePasswordAuthenticationToken auth = new UsernamePasswordAuthenticationToken(login, null, foundUser
-                        .get()
-                        .getAuthorities());
-                return auth;
+        try {
+            final DecodedJWT decodedJwt = JWT.require(Algorithm.HMAC256(modelPropertiesService.securitySecret.getBytes()))
+                    .build()
+                    .verify(tokenWithoutPrefix);
+            final String login = decodedJwt.getSubject();
+            if (login != null && isNotLoggedOut(tokenWithoutPrefix)) {
+                final Optional<User> foundUser = userRepository.findUserByLogin(login);
+                if (foundUser.isPresent()) {
+                    final UsernamePasswordAuthenticationToken auth = new UsernamePasswordAuthenticationToken(login, null, foundUser
+                            .get()
+                            .getAuthorities());
+                    return auth;
+                }
+                return null;
```

Comment:
@syphzur why returm here whem ur return next line anyway

Reply:
because i need to return here

Reply:
no